### PR TITLE
increase _include capability

### PR DESCRIPTION
**Pull request by elsiehoffet-94**
_include only added the last input to the bundle, now it adds all included resources

### CODE REVIEW

File: fhirstore/fhirstore.py [hunk 1]
```diff
@@ -409,7 +409,7 @@ def comprehensive_search(self, resource_type: str, args: ImmutableMultiDict):
                             f"{e.info['error']['index']} is not indexed in the database yet."
                         )
 
-            bundle.append(included_hits, search_args.formatting_args)
+                        bundle.append(included_hits, search_args.formatting_args)
```
Comment (Jasopaum) [inline fhirstore/fhirstore.py:412]: I don't think this should be in the `except`

Comment (simonvadee) [reply fhirstore/fhirstore.py:412]: true

File: fhirstore/fhirstore.py [hunk 2]
```diff
@@ -382,34 +381,33 @@ def comprehensive_search(self, resource_type: str, args: ImmutableMultiDict):
...
+                include_refs = {} 
+                try:
+                    for item in bundle.content["entry"]:
+                        if item["search"]["mode"] == "match":
+                            ref_to_parse = item["resource"][attribute]["reference"].split(sep="/", maxsplit=1)
+                            if ref_to_parse[0] not in include_refs :
+                               include_refs[ref_to_parse[0]] = set()
+                            include_refs[ref_to_parse[0]].add(ref_to_parse[1])
+                    for included_resource in include_refs.keys():
+                        include_core_query = build_core_query({"multiple":{"id": include_refs[included_resource]}})
```
Comment (Jasopaum) [inline fhirstore/fhirstore.py:394]: You can write
```
for included_resource, included_ids in include_refs.items():
    include_core_query = build_core_query({"multiple":{"id": included_ids}})
```

File: fhirstore/fhirstore.py [hunk 3]
```diff
@@ -382,34 +381,33 @@ def comprehensive_search(self, resource_type: str, args: ImmutableMultiDict):
...
-                            "reference"
-                        ].split(sep="/", maxsplit=1)
+            for attribute in search_args.formatting_args["include"]:
+                include_refs = {} 
+                try:
+                    for item in bundle.content["entry"]:
+                        if item["search"]["mode"] == "match":
+                            ref_to_parse = item["resource"][attribute]["reference"].split(sep="/", maxsplit=1)
+                            if ref_to_parse[0] not in include_refs :
+                               include_refs[ref_to_parse[0]] = set()
```
Comment (Jasopaum) [inline fhirstore/fhirstore.py:391]: You could use a `defaultdict(set)` to avoid this check

File: fhirstore/fhirstore.py [hunk 4]
```diff
@@ -382,34 +381,33 @@ def comprehensive_search(self, resource_type: str, args: ImmutableMultiDict):
...
-                    continue
-                # For each attribute to include
-                for attribute in search_args.formatting_args["include"]:
-                    # split the reference attribute "Practioner/123" into a
-                    # resource "Practioner" and an id "123"
-                    try:
-                        included_resource, included_id = item["resource"][attribute][
-                            "reference"
-                        ].split(sep="/", maxsplit=1)
+            for attribute in search_args.formatting_args["include"]:
```
Comment (simonvadee) [inline fhirstore/fhirstore.py:384]: for .. try .. for .. if .. for ? meh OK

**Pull request merged**